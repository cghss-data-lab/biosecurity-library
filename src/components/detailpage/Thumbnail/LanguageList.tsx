import Tippy from '@tippyjs/react'
import React from 'react'
import styled from 'styled-components'

interface LanguageListProps {
  languages: string[] | undefined
  languageFiles: {
    [key: string]: string
  }
  languageURLs: {
    [key: string]: string
  }
}

const ListParagraph = styled.p`
  padding-top: 1em;
`

const LanguageList = ({
  languages,
  languageFiles,
  languageURLs,
}: LanguageListProps) => {
  if (!languages || languages.length === 0)
    languages =
      Object.keys(languageFiles).length > 0
        ? Object.keys(languageFiles)
        : Object.keys(languageURLs)

  let list = ''
  if (languages.length === 1) list = languages[0]
  if (languages.length <= 3) list = languages.slice(0, 3).join(', ')
  if (languages.length > 3) list = languages.slice(0, 3).join(', ') + '...'

  const tooltip = (
    <div>
      {languages.map(lang => (
        <div style={{ padding: 5 }}>{lang}</div>
      ))}
    </div>
  )

  if (languages.length > 3)
    return (
      <Tippy content={tooltip} theme={'light'} placement={'bottom-end'}>
        <ListParagraph>
          {languages.length === 1 ? 'Language: ' : 'Languages: '} {list}{' '}
          {languages.length > 3 && `(${languages.length})`}
        </ListParagraph>
      </Tippy>
    )

  return (
    <ListParagraph>
      {languages.length === 1 ? 'Language: ' : 'Languages: '} {list}{' '}
      {languages.length > 3 && `(${languages.length})`}
    </ListParagraph>
  )
}

export default LanguageList
