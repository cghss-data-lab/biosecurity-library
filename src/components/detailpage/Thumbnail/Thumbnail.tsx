import React from 'react'
import styled from 'styled-components'

import { GatsbyImage, getImage, ImageDataLike } from 'gatsby-plugin-image'
import DownloadOrAccessDropdown from '../DownloadOrAccessDropdown/DownloadOrAccessDropdown'

const ThumbnailContainer = styled.div`
  grid-area: thumbnail;
  display: flex;
  flex-direction: column;

  @media (min-width: 1000px) {
    display: block;
    padding: 30px 15px 15px 0;
    margin-top: 20px;
  }
`

const ThumbnailShadow = styled(GatsbyImage)`
  box-shadow: 0px 5px 10px rgba(0, 0, 0, 0.24);
  border-radius: 5px;
`

interface ThumbnailProps {
  data: {
    Thumbnail_INTERNAL: {
      localFiles: ImageDataLike[]
    }
    Short_name: string
    Other_language_URLs: string
    URL_for_resource: string
    Resource_language?: string[]
    Primary_file?: {
      localFiles: {
        publicURL: string
        name: string
      }[]
    }
    Other_language_files_INTERNAL?: {
      localFiles: {
        publicURL: string
        name: string
      }[]
    }
  }
}

const Thumbnail = ({ data }: ThumbnailProps): JSX.Element => {
  const thumbnail = getImage(data.Thumbnail_INTERNAL?.localFiles[0])

  if (thumbnail) {
    return (
      <ThumbnailContainer>
        {thumbnail && (
          <ThumbnailShadow
            image={thumbnail}
            alt={data.Short_name + 'thumbnail image'}
            style={{
              boxShadow: '0px 5px 10px rgba(0, 0, 0, 0.24)',
              margin: 'auto',
            }}
          />
        )}
        <DownloadOrAccessDropdown showLanguages {...{ data }} />
      </ThumbnailContainer>
    )
  }
  return <ThumbnailContainer />
}

export default Thumbnail
