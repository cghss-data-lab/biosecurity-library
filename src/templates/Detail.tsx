import React from 'react'
import { ImageDataLike } from 'gatsby-plugin-image'
import * as network from '@talus-analytics/viz.charts.network'

// layout components
import Main from '../components/layout/Main'
import NavBar from '../components/layout/NavBar/NavBar'
import FigmaProvider from '../figma/FigmaProvider'

// page-specific components
import Breadcrumbs from '../components/detailpage/Breadcrumbs/Breadcrumbs'
import Grid from '../components/detailpage/Grid'
import Header from '../components/detailpage/Header/Header'
import Thumbnail from '../components/detailpage/Thumbnail/Thumbnail'
import TabSection from '../components/detailpage/TabSection/TabSection'

/**
 * Resource Airtable fields that link to other records
 */
export type LinkField = 'Auto_other_resources_cited'
export type ResourceSetProps = {
  data: {
    Resource_set_name: string
    Unique_ID: number
    Description: string | null
    Resources_in_set: {
      data: {
        Record_ID_INTERNAL: string
        Resource_name: string
        Short_name: string
        Resource_type: string
      }
    }[]
  }
}
export interface PageContext {
  data: {
    Record_ID_INTERNAL: string
    Short_description: string
    Long_description: string
    Key_topic_area: string[]
    Resource_name: string
    Short_name: string
    Resource_type: string
    Authoring_organization: {
      data: {
        Name: string
      }
    }[]
    Target_user_role: string[]
    Potential_user_role: string[]
    URL_for_resource: string
    Access_information: string
    Access_limitations: string
    Resource_language: string[]
    Edition: string | null
    First_release_date: string
    Last_update_date: string
    Update_frequency: string
    // Topic_area_icons: string
    Auto_other_resources_cited:
      | {
          data: {
            Record_ID_INTERNAL: string
          }
        }[]
      | null
    // Topic_area_icons: string
    Files_INTERNAL: {
      localFiles: {
        publicURL: string
        name: string
      }[]
    }
    Thumbnail_INTERNAL: {
      localFiles: ImageDataLike[]
    }
    Resource_sets: ResourceSetProps[] | null

    // Below: Additional data elements not directly sourced from Airtable.

    /**
     * Optional: Resource map nodes/links data to render in a network map on the
     * resource's page. Defaults to undefined (no map rendered).
     */
    resourceMapData?: network.AppGraphData
  }
}

const Detail: React.FC<{ pageContext: PageContext }> = ({
  pageContext: { data },
}) => (
  <FigmaProvider>
    <NavBar />
    <Main>
      <Breadcrumbs {...{ data }} />
      <Grid>
        <Thumbnail {...{ data }} />
        <Header {...{ data }} />
        <TabSection {...{ data }} />
      </Grid>
    </Main>
  </FigmaProvider>
)

export default Detail
