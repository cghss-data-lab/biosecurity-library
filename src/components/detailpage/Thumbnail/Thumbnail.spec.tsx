import React from 'react'
import { render } from '@testing-library/react'
import { PageContext } from '../../../templates/Detail'
import Thumbnail from './Thumbnail'

describe('Thumbnail', () => {
  it('renders correctly', () => {
    const result = render(<Thumbnail data={MOCK_RESOURCE_DATA} />)
    expect(result).toMatchSnapshot()
  })
})

const MOCK_RESOURCE_DATA: PageContext['data'] = {
  Short_description: 'Test',
  Long_description: 'Test',
  Key_topic_area: ['Test'],
  Resource_name: 'Test',
  Short_name: 'Test',
  Resource_type: 'Test',
  Authoring_organization: 'Test',
  Target_user_role: ['Test'],
  Potential_user_role: ['Test'],
  URL_for_resource: 'Test',
  Access_method: 'Test',
  Access_limitations: 'Test',
  Resource_language: ['Test'],
  Edition: 'Test',
  First_release_date: 'Test',
  Last_update_date: 'Test',
  Update_frequency: 'Test',
  Topic_area_icons: 'Test',
  Files_INTERNAL: {
    localFiles: [{ publicURL: 'Test', name: 'Test' }],
  },
  Thumbnail_INTERNAL: {
    localFiles: [],
  },
  Auto_other_resources_cited: ['Test 2'],
}
