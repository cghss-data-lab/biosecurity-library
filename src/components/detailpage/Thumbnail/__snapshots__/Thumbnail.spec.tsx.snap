// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail renders correctly 1`] = `
<div
  className="sc-bdfBQB kwGLrC"
/>
`;
